cmake/CMakeLists.txt

```diff
@@ -264,9 +265,19 @@ if (NOT MSVC AND NOT onnxruntime_ENABLE_BITCODE)
   string(APPEND CMAKE_C_FLAGS " -ffunction-sections -fdata-sections")
 endif()
 
-# Build WebAssembly source map on debug build.
 if (onnxruntime_BUILD_WEBASSEMBLY)
-  string(APPEND CMAKE_CXX_FLAGS_DEBUG " -g4 --source-map-base http://localhost:8000/")
+  # Enable LTO for release build
+  if (NOT CMAKE_BUILD_TYPE STREQUAL "Debug")
+    string(APPEND CMAKE_CXX_FLAGS " -flto")
```

Comment:
But we already have a cmake option for this?

Reply:
yes, but it looks like `set (CMAKE_INTERPROCEDURAL_OPTIMIZATION ON)` does not work correctly on emcc.

do you suggest to append flags at https://github.com/microsoft/onnxruntime/blob/73a6af52db6fd14c7e0de918ba12618fb582a5d0/cmake/CMakeLists.txt#L341

Reply:
I think I can utilize this flag. i am doing some local testing and will update this part once done

Reply:
after some tests, I decide do not use this flag, because it sets "-flto=thin", which is not supported by wasm-ld.